src/utils/ProfileUtils.ts

```diff
@@ -180,17 +180,23 @@ export async function populateImportedProfile(
     }
 }
 
-//TODO: Check if instead of returning null/empty strings, there's some errors that should be handled
-export async function readProfileFile(file: string) {
-    let read = '';
+export async function readProfileFile(file: string): Promise<string> {
+    let read: string | undefined;
     if (file.endsWith('.r2x')) {
         read = (await FsProvider.instance.readFile(file)).toString();
     } else if (file.endsWith('.r2z')) {
-        const result: Buffer | null = await ZipProvider.instance.readFile(file, "export.r2x");
-        if (result === null) {
-            return null;
-        }
-        read = result.toString();
+        await ZipProvider.instance.readFile(file, "export.r2x").then(
+            (value) => { read = value?.toString() },
+            () => { return null; }
+        );
+    }
+
+    if (!read) {
+        throw new R2Error(
+            'Error when reading file contents',
+            'Reading the .r2z file contents failed. File contents be empty or corrupted.',
```

Comment:
> Reading the .r2z file contents failed.
It's not really the r2z that's the problem, but r2x?

> File contents be empty or corrupted.
This sounds a bit "street" way to phrase it? 😂 Should it be "are" instead of "be"?